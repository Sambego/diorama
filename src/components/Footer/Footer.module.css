.footer {
  position: fixed;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 1;
  display: flex;
  justify-content: space-between;
  width: 100%;
  padding: 2rem 4rem;
  font-size: 1.5vw;
  font-weight: 700;
}

.left {
  text-alignn: left;
}

.right {
  margin-left: auto;
  text-align: right;
}
