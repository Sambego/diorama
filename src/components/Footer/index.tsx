import React from "react";
import PropTypes from "prop-types";

import styles from "./Footer.module.css";

type FooterProps = {
	className?: string;
	style?: React.CSSProperties;
	left?: React.ReactNode;
	right?: React.ReactNode;
};

function Footer({ left, right, style, className }: FooterProps) {
	return (
		<footer
			style={style}
			className={`${styles.footer} diorama-footer ${className}`}
		>
			{left && (
				<div className={`${styles.left} diorama-footer-left`}>{left}</div>
			)}
			{right && (
				<div className={`${styles.right} diorama-footer-right`}>{right}</div>
			)}
		</footer>
	);
}

Footer.propTypes = {
	left: PropTypes.node,
	right: PropTypes.node,
	className: PropTypes.string,
	style: PropTypes.shape({}),
};

Footer.defaultProps = {
	left: undefined,
	right: undefined,
	className: "",
	style: {},
};

export default Footer;
