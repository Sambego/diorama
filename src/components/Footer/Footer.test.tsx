import React from "react";
import renderer from "react-test-renderer";
import Footer from ".";

it("renders correctly", () => {
	const tree = renderer.create(<Footer left="@sambego" />).toJSON();
	expect(tree).toMatchSnapshot();
});
