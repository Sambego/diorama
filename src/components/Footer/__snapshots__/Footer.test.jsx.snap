// Vitest Snapshot v1

exports[`renders correctly 1`] = `
<footer
  className="_footer_gkmlu_1 diorama-footer "
  style={{}}
>
  <div
    className="_left_gkmlu_15 diorama-footer-left"
  >
    @sambego
  </div>
</footer>
`;
